### PR TITLE
Fix: useTreeState rebuilds the tree collection when a node is expanded.

### LINKED ISSUE #1525: useTreeState doesn't rebuild the tree collection when a node is expanded
# 🐛 Bug Report 

When `useTreeState` is used in an uncontrolled component, it doesn't rebuild the tree collection after the internal `expandedKeys` state changes due to a `toggleKey()` call.

As a result, the `prevKey` and `nextKey` props of the children of the newly expanded node never get set, which prevents keyboard navigation from working on those children when a keyboard delegate is attached to the tree via `useSelectableCollection`.

## 🤔 Expected Behavior

In an uncontrolled component, an internal change to the `expandedKeys` state should cause the tree collection to be rebuilt, so that the `prevKey` and `nextKey` props of each child node are updated.

## 😯 Current Behavior

The tree collection is not rebuilt when `expandedKeys` state changes internally.

## 💁 Possible Solution

I think this can be fixed by changing this line in `useTreeState`:

https://github.com/adobe/react-spectrum/blob/c66852948e2eb7427ce24d4f3199e53c4f615919/packages/%40react-stately/tree/src/useTreeState.ts#L54

...to:

```typescript
let tree = useCollection(props, nodes => new TreeCollection(nodes, {expandedKeys}), null, [expandedKeys]);
```

Happy to send a PR if there's agreement!

## 🔦 Context

I'm building a treeview component that uses `useTreeState` to manage the state of the tree and `useSelectableCollection` to manage selection state and keyboard interactions within the tree.

## 💻 Code Sample

Reduced test case: https://codepen.io/rgrove/pen/eYBNpao

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| \@react-stately/tree   | 3.1.2
| Browser          | n/a
| Operating System | n/a

## 🧢 Your Company/Team

[SmugMug](https://www.smugmug.com/)

### PR DESCRIPTION
**Pull request by lukasbuenger**
Fixes #1525

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues/1525).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [x] Updated documentation (if it already exists for this component).
- [x] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

I could have extended the Tree stories but in order to test this rather small fix I would have had to extend it quite a lot and given how bare bones those stories still are, I wasn't sure how to stay within your guidelines/style.  

## 🧢 Your Project:

At this point I'm really just playing around with the libraries.

### CODE REVIEW

**Review by LFDanLu — COMMENTED** [review]
Apologies for the delay and thanks for the fix. I'll push a story to test this soon.

File: packages/@react-stately/tree/stories/useTreeState.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+import {useSelectableCollection, useSelectableItem} from '@react-aria/selection';
+import {useTreeState} from '../';
+
+storiesOf('useTreeState', module)
+  .add(
+    'keyboard navigation',
+    () => render()
+  );
+
+function render(props = {}) {
```
Comment (LFDanLu) [inline packages/@react-stately/tree/stories/useTreeState.stories.tsx:28]: story largely sourced from original issue submitter (thanks @rgrove!)

**Review by LFDanLu — APPROVED** [review]

**Review by snowystinger — DISMISSED** [review]
Looks like it's working as described now

**Review by LFDanLu — APPROVED** [review]

**Review by snowystinger — APPROVED** [review]
LGTM

**Pull request merged**